src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/MarkdownContentHandler.cs

```diff
@@ -66,50 +92,80 @@ protected override void HandleCurrentProperty(object declaringObject, PropertyIn
                 }
             }
 
-            protected override PropInfo[] GetProps(Type type)
+            protected override void HandleDictionaryType(object declaringObject, PropertyInfo currentPropertyInfo, HandleModelAttributesContext context)
             {
-                return (from prop in ReflectionHelper.GetSettableProperties(type)
-                        let attr = prop.GetCustomAttribute<MarkdownContentAttribute>()
-                        select new PropInfo
+                var type = currentPropertyInfo.PropertyType;
+                if (ReflectionHelper.ImplementsGenericDefintion(type, typeof(IDictionary<,>)))
+                {
+                    dynamic dict = currentPropertyInfo.GetValue(declaringObject);
+                    if (dict != null && dict.Count > 0)
+                    {
+                        var keys = new List<dynamic>(dict.Keys);
```

Comment:
>new List<dynamic>(dict.Keys); [](start = 35, length = 29)

question: why not enumerate dictionary directly? is it to prevent dictionary to be modified?